lib/react.dart

```diff
@@ -590,6 +591,28 @@ abstract class Component2 implements Component {
     adapter.setStateWithUpdater(updater, callback);
   }
 
+  /// Initializes this component's [state] to [value].
+  ///
+  /// Whereas `this.state` can be set directly in JavaScript components during initialization,
+  /// supporting that is difficult in the Dart implementation, so this method must be used instead.
+  ///
+  /// __Example__:
+  ///     ```js
```

Comment:
I don't think ``` is necessary / works to output code blocks in dart doc comments.  Should just be:

```
  /// __Example__:
  ///
  ///     // JavaScript
  ///     constructor() {
  ///       this.state = {count: 0};
  ///     }
  ///
  ///     // Dart
  ///     init() {
  ///       initializeState({'count': 0});
  ///     }
```